src/node/utils/UpdateCheck.js

```diff
@@ -2,23 +2,35 @@
 const semver = require('semver');
 const settings = require('./Settings');
 const axios = require('axios');
+const headers = {
+  'User-Agent': 'Etherpad/' + settings.getEpVersion(),
+}
+
+const updateInterval = 60 * 60 * 1000; // 1 hour
 let infos;
+let lastLoadingTime = null;
+
+const loadEtherpadInformations = () => {
+  if (lastLoadingTime !== null && Date.now() - lastLoadingTime < updateInterval) {
+    return Promise.resolve(infos);
+  }
 
-const loadEtherpadInformations = () =>
-    axios.get('https://static.etherpad.org/info.json')
-        .then(async resp => {
-          try {
-            infos = await resp.data;
-            if (infos === undefined || infos === null) {
-              await Promise.reject("Could not retrieve current version")
-              return
-            }
-            return await Promise.resolve(infos);
-          }
-          catch (err) {
-            return   await Promise.reject(err);
-          }
-        })
+  return axios.get('https://static.etherpad.org/info.json', {headers: headers})
+  .then(async resp => {
```

Comment:
axios has a separate method for handling error. Everything that is not 2xx like 404, 500, etc. are handled in .catch after .then. There you can put the error handling measures.

Reply:
I reformatted the code and have not changed the exception handling, but you're right. I improved the exception handling. Please have a look again 🙂

Thanks, but don't expect too much. Just wanted to improve a little bit how my server is getting called 😅

Reply:
Nice. I'll take a look asap and check if it's working. 

Ah okay. Thought I might get somebody back to the project 😂.